oneflow/core/common/stride.cpp

```diff
@@ -29,6 +31,38 @@ Stride::Stride(const Shape& shape) {
   }
 }
 
+Stride::Stride(const std::shared_ptr<Shape>& shape) {
+  if (shape->NumAxes() > 0) {
+    stride_vec_.resize(shape->NumAxes());
+    int64_t stride = 1;
+    for (size_t i = shape->NumAxes(); i > 0; --i) {
+      stride_vec_.at(i - 1) = stride;
+      stride *= shape->At(i - 1);
+    }
```

Comment:
这个逻辑在 c++17 可以改成
```suggestion
    std::exclusive_scan(shape->rbegin(), shape->rend(), stride_vec_->begin(), 1, std::multiplies<>{});
```
更加容易理解和不会写错。

我实现了一版 c++14 的，等服务器恢复了上传上去

Reply:
已经合并到 master 了，include "oneflow/core/common/cplusplus_17.h" 就可以

Reply:
已修改